### PR TITLE
Reporter web app updates

### PR DESCRIPTION
**Pull request by tsteenbe**
See individual commit message for details

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1284)

### CODE REVIEW

File: reporter-web-app/src/reducers/selectors.js [hunk 1]
```diff
@@ -73,8 +73,8 @@ const getTotalNrLicenses = (projectsLicenses) => {
         return accumulator;
     }, licensesSet).size;
 };
-const hasReportDataChanged = (newArg, lastArg) => newArg.data.reportLastUpdate === lastArg.data.reportLastUpdate;
-
+const hasReportDataChanged = (newArgs, oldArgs) => newArgs.length === oldArgs.length
+    && newArgs[0].data.reportLastUpdate === oldArgs[0].data.reportLastUpdate;
```
Comment (sschuberth) [inline reporter-web-app/src/reducers/selectors.js]: Isn't the logic inverted here? The variable is called "has changed", but you check for equality here.

File: reporter-web-app/config/env.js [hunk 2]
```diff
@@ -42,12 +42,12 @@ dotenvFiles.forEach(dotenvFile => {
 
 // We support resolving modules according to `NODE_PATH`.
 // This lets you use absolute paths in imports inside large monorepos:
-// https://github.com/facebookincubator/create-react-app/issues/253.
+// https://github.com/facebook/create-react-app/issues/253.
```
Comment (sschuberth) [inline reporter-web-app/config/env.js]: Commit message: "Change based dependencies", I don't get that sentence. Are you missing an "on"?

File: reporter-web-app/scripts/build.js [hunk 3]
```diff
@@ -14,11 +14,13 @@ process.on('unhandledRejection', err => {
 // Ensure environment variables are read.
 require('../config/env');
 
+
```
Comment (sschuberth) [inline reporter-web-app/scripts/build.js]: Nit: Blank line added.

File: reporter-web-app/scripts/start.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ process.on('unhandledRejection', err => {
 // Ensure environment variables are read.
 require('../config/env');
 
+
```
Comment (sschuberth) [inline reporter-web-app/scripts/start.js]: Nit: Blank line added.

File: reporter-web-app/scripts/start.js [hunk 5]
```diff
@@ -54,18 +55,27 @@ if (process.env.HOST) {
   console.log(
     `If this was unintentional, check that you haven't mistakenly set it in your shell.`
   );
-  console.log(`Learn more here: ${chalk.yellow('http://bit.ly/2mwWSwH')}`);
+  console.log(
```
Comment (sschuberth) [inline reporter-web-app/scripts/start.js]: Shouldn't these be combined to a single `log` call with `\n` n between?

File: reporter-web-app/scripts/test.js [hunk 6]
```diff
@@ -15,12 +15,38 @@ process.on('unhandledRejection', err => {
 // Ensure environment variables are read.
 require('../config/env');
 
+
```
Comment (sschuberth) [inline reporter-web-app/scripts/test.js]: Nit: Blank line added.

File: reporter-web-app/package.json [hunk 7]
```diff
@@ -17,6 +17,7 @@
         "memoize-one": "^5.0.0",
         "prop-types": "^15.6.2",
         "react": "^16.7.0",
+        "react-app-polyfill": "^0.2.0",
```
Comment (sschuberth) [inline reporter-web-app/package.json]: Commit message nit: Please end sentences with a dot.

File: reporter-web-app/config/webpack.config.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 const fs = require('fs');
+const fse = require('fs-extra');
```
Comment (sschuberth) [inline reporter-web-app/config/webpack.config.js]: Commit message nit: s/Adds/Add/. End sentence with a dot.

File: reporter-web-app/package.json [hunk 9]
```diff
@@ -30,7 +30,6 @@
         "spdx-license-list": "^5.0.0"
     },
     "scripts": {
-        "analyze": "source-map-explorer build/static/js/main.*",
```
Comment (sschuberth) [inline reporter-web-app/package.json]: Commit message nit: s/Removing/Remove/. End sentence with a dot.

File: reporter-web-app/README.md [hunk 10]
```diff
@@ -1,175 +1,18 @@
-This project was bootstrapped with [Create React App](https://github.com/facebookincubator/create-react-app).
+This directory contains the code for Web App report generated by the ORT reporter.
```
Comment (sschuberth) [inline reporter-web-app/README.md]: Commit message nits: s/includes/include/. End sentence with a dot.

File: reporter-web-app/src/App.css [hunk 11]
```diff
@@ -249,3 +254,8 @@ ul.ort-table-list {
...
...-12,34.5-31.8,34.5c-19.7,0-31.8-12.7-31.8-34.6c0-20.8,12-34.4,31.8-34.4 C885.7,159,897.7,171.7,897.7,193.6z M854.2,193.5c0,14.4,3.9,20.4,11.7,20.4s11.7-6.2,11.7-20.3c0-14.4-3.9-20.4-11.7-20.4 S854.2,179.4,854.2,193.5z'/%3e%3cpath class='st95' d='M969.1,193.6c0,20.8-12,34.5-31.8,34.5c-19.7,0-31.8-12.7-31.8-34.6c0-20.8,12-34.4,31.8-34.4 C957.1,159,969.1,171.7,969.1,193.6z M925.6,193.5c0,14.4,3.9,20.4,11.7,20.4s11.7-6.2,11.7-20.3c0-14.4-3.9-20.4-11.7-20.4 S925.6,179.4,925.6,193.5z'/%3e%3cpath class='st95' d='M979.6,209.5v-74.2l19.3-2.1v75.5c0,2.7,1.1,4,3.4,4c1.2,0,2.3-0.2,3.2-0.6l3.7,13.7c-2.8,1.2-7,2.2-11.7,2.2 C986.2,228.1,979.6,221.3,979.6,209.5z'/%3e%3cpath class='st95' d='M1036.4,226h-19.3v-90.6l19.3-2.1V226z M1056.8,189.6L1078,226h-21.7l-19.3-35.4l19.8-29.4h19.7L1056.8,189.6 z'/%3e%3cpath class='st95' d='M1105.4,137.9c0,6.2-4.8,11-11.5,11s-11.4-4.8-11.4-11c0-6.2,4.6-11,11.4-11S1105.4,131.7,1105.4,137.9z M1103.7,226h-19.3v-64.9h19.3V226z'/%3e%3cpath class='st95' d='M1159.4,222.9c-4.9,3.4-11.7,5.1-17.6,5.1c-14.3-0.1-21.3-8.2-21.3-23.5v-30.1h-9.2v-13.4h9.2v-14l19.3-2.2 v16.2h14.9l-2.1,13.4h-12.8v29.8c0,6.2,2.1,8.4,6.1,8.4c2.2,0,4.4-0.6,7-2.1L1159.4,222.9z'/%3e%3c/g%3e%3c/svg%3e ");
```
Comment (sschuberth) [inline reporter-web-app/src/App.css]: I don't like that we inline the SVG here. Instead, we should commit an SVG file and add code to generate this from the SVG file.

File: reporter-web-app/package.json [hunk 12]
```diff
@@ -51,8 +51,8 @@
         "dotenv": "6.0.0",
         "dotenv-expand": "4.2.0",
         "eslint": "5.6.0",
-        "eslint-config-react-app": "^3.0.6",
         "eslint-config-airbnb": "17.1.0",
+        "eslint-config-react-app": "^3.0.6",
```
Comment (sschuberth) [inline reporter-web-app/package.json]: Commit message nit: Write out `devDependencies`.

File: reporter-web-app/src/components/SummaryViewLicenses.js [hunk 13]
```diff
@@ -19,7 +19,7 @@
 
 import React from 'react';
 import {
-    Col, Table, Tabs, Tag, Row
+    Col, Icon, Table, Tabs, Row
```
Comment (sschuberth) [inline reporter-web-app/src/components/SummaryViewLicenses.js]: Commit message: s/redering/rendering/. s/word wrap/word-wrap/.

File: reporter-web-app/src/components/SummaryViewLicenses.js [hunk 14]
```diff
@@ -79,9 +79,10 @@ const SummaryViewLicenses = (props) => {
                 sortOrder: sortedInfo.columnKey === 'name' && sortedInfo.order,
                 key: 'name',
                 render: (text, row) => (
-                    <Tag color={row.color}>
-                        {text}
-                    </Tag>
+                    <span>
+                        <Icon type="file-text" style={{ color: row.color }} />
+                        {` ${text}`}
```
Comment (sschuberth) [inline reporter-web-app/src/components/SummaryViewLicenses.js]: Why is this not just `{text}` as before? Is the leading space correct here? Shouldn't we rather use a right-side margin for the icon if you want to add some spacing between icon and text?

File: reporter-web-app/src/reducers/index.js [hunk 15]
```diff
@@ -87,8 +87,8 @@ export default (state = initState, action) => {
                 ...state.app,
                 loading: {
                     ...state.app.loading,
-                    text: 'Processing done...',
-                    percentage: 100
+                    text: 'Processing report data done...',
```
Comment (sschuberth) [inline reporter-web-app/src/reducers/index.js]: Please reword the commit message body text, the grammar is a bit weird 😉

File: reporter-web-app/src/reducers/selectors.js (hunk 1 above)
Comment (tsteenbe) [reply reporter-web-app/src/reducers/selectors.js]: Let me investigate

File: reporter-web-app/config/env.js (hunk 2 above)
Comment (tsteenbe) [reply reporter-web-app/config/env.js]: Env.js comes directly from create-react-app - will need to fill a PR with upstream to get this fixed

File: reporter-web-app/scripts/start.js (hunk 5 above)
Comment (tsteenbe) [reply reporter-web-app/scripts/start.js]: start.js comes from upstream create-react-app

File: reporter-web-app/src/App.css (hunk 11 above)
Comment (tsteenbe) [reply reporter-web-app/src/App.css]: We commit the SVG file inside of the CSS as the build output is a single HTML file. Was looking at WebPack plugins to do this such as https://github.com/webpack-contrib/svg-inline-loader but could not get them to work with version of WebPack we are using. Happy to file an improvement ticket for this to look at this at a later time.

File: reporter-web-app/scripts/start.js (hunk 4 above)
Comment (tsteenbe) [reply reporter-web-app/scripts/start.js]: File comes from upstream create-react-app so prefer not to modify it

File: reporter-web-app/src/reducers/selectors.js (hunk 1 above)
Comment (tsteenbe) [reply reporter-web-app/src/reducers/selectors.js]: Fixed it in https://github.com/heremaps/oss-review-toolkit/pull/1284/commits/73940e542fe2d9f79431ecf63cadfa47c98f8495

File: reporter-web-app/scripts/build.js (hunk 3 above)
Comment (tsteenbe) [reply reporter-web-app/scripts/build.js]: File comes from upstream create-react-app so prefer not to modify it

**Pull request merged**